### ISSUE TITLE
Issue with no distance and not clean

### ISSUE THREAD

**Issue by awiddersheim** [pypa/setuptools-scm#293]
The documentation says this in regards to behavior:

```
no distance and not clean:
   {tag}+dYYYMMMDD
```

However, with `git` at least, I end up with the `distance and not clean` scenario:

```
{next_version}.dev{distance}+{scm letter}{revision hash}.dYYYMMMDD
```

Haven't tried anything other than `git`. It looks like my `distance` is always `0` but this logic [here](https://github.com/pypa/setuptools_scm/blob/master/src/setuptools_scm/version.py#L151) is looking for `None` which means [this](https://github.com/pypa/setuptools_scm/blob/master/src/setuptools_scm/version.py#L256) never works.

Tried a few code changes but had some trouble with getting tests to pass and run without getting mercurial setup and what not.

**Comment by awiddersheim** [comment]
Hrm, looking at the tests, these seem to contradict what is in the docs when it comes to expected behavior:

```
def test_version_from_git(wd):
    assert wd.version == "0.1.dev0"

    wd.commit_testfile()
    assert wd.version.startswith("0.1.dev1+g")
    assert not wd.version.endswith("1-")

    wd("git tag v0.1")
    assert wd.version == "0.1"

    wd.write("test.txt", "test2")
    assert wd.version.startswith("0.2.dev0+g") <--- Should be no distance and not clean ({tag}+dYYYMMMDD)?

    wd.commit_testfile()
    assert wd.version.startswith("0.2.dev1+g") <--- This looks okay as it's distance and clean ({next_version}.dev{distance}+{scm letter}{revision hash}). Might be worth validating no date was added at the end though.
```

**Comment by awiddersheim** [comment]
Thought about this a bit more and it seems like over time the behavior has changed to really be:

```
no commits and not clean:
   {tag}+dYYYMMMDD
```

Which makes sense since you can't add a git tag without any commits. This is also how it functions today so things have maybe just evolved to this despite parts of the logic and the documentation suggesting otherwise.

**Comment by RonnyPfannschmidt** [comment]
thanks for bringing together the details, unfortunately i wont be able to dig deeper into it until i released 3.0.4 and fixed the build pipeline

**Comment by RonnyPfannschmidt** [comment]
fixed in f6d8be8caff12d4a9e35c91aada14841cdbcae86
